src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -305,6 +333,18 @@ private static CodeAction GetNestedActionsFromActionSet(IUnifiedSuggestedAction
                 codeAction.Title, nestedActions.ToImmutable(), codeAction.IsInlinable, codeAction.Priority);
         }
 
+        private static CodeAction GetFixAllActionsFromActionSet(IUnifiedSuggestedAction suggestedAction, string? fixAllScope)
+        {
+            var codeAction = suggestedAction.OriginalCodeAction;
+            if (suggestedAction is not UnifiedCodeFixSuggestedAction unifiedCodeFixSuggestedAction || unifiedCodeFixSuggestedAction.FixAllFlavors is null)
+            {
+                return codeAction;
+            }
+
+            var fixAllFlavor = unifiedCodeFixSuggestedAction.FixAllFlavors.Actions.OfType<UnifiedFixAllCodeFixSuggestedAction>().Where(action => action.FixAllState.Scope.ToString() == fixAllScope).First();
```

Comment:
is this guaranteed to succeed?  is there any chance that a different set of code actions was computed when we received the resolve callback?  or are we guaranteed to have the original set?  please doc.

Reply:
We should be guaranteed to have the original set I believe, but will doc.